src/Azure/Orleans.Transactions.AzureStorage/Storage/AzureTransactionLogStorage.cs

```diff
@@ -322,6 +357,30 @@ private List<CommitRecord> DeserializeCommitRecords(byte[] serializerCommitRecor
             return commitRecords;
         }
 
+        public class ArchivalRow : TableEntity
+        {
+            public ArchivalRow()
+            {
+            }
+
+            public ArchivalRow(byte[] transactions, long firstTransactionId, long firstLSN, string clusterId, DateTime archivalTime)
+            {
+                this.Transactions = transactions;
+                this.ArchivalTime = archivalTime;
+                this.ClusterId = clusterId;
+                this.FirstLSN = firstLSN;
+                this.FirstTransactionId = firstTransactionId;
+                PartitionKey = $"{clusterId}${this.ArchivalTime.Ticks}";
```

Comment:
Probably should have delimiter between cluster id and archive time, for parsablity.
Ticks format should probably be normalized, suggest 16:x

Reply:
Suggest formal row key structure something like:
        private const string  RKBaseFormat "arch_{0}_";
        public static string RowKeyMin(string clusterId) => string.Format(RKBaseFormat,   clusterId);
        public static string RowKeyMax(string clusterId) => RowKeyMin(clusterId) + "~";
        public static string MakeRowKey(string clusterId, DateTime utcNow) => RowKeyMin(clusterId) +utcNow.Ticks.ToString("{0:x16}");

That way queries can be formed from the min to max row keys without knowledge or concern as to what those values are.

Reply:
I didn't read close enough.  Thought this was the row key.  Partition key should probably match log record partition key, and all of the above is suggested for the row key.  Also, data partition needs be by service id, rather than cluster Id.